roles/basic_app_examples/nginx_backup_restore/tasks/check.yml

```diff
@@ -1,7 +1,7 @@
 - name: Wait 2 minutes for nginx deployment
   k8s_facts:
     kind: Deployment
-    api_version: apps/v1
+    api_version: apps/v1beta1
```

Comment:
You may want to set this based on Kubernetes version

Reply:
It was created based on https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#deployment-v1-apps and https://docs.openshift.com/container-platform/3.11/rest_api/apis-apps/v1beta1.Deployment.html which a basically the same.

Reply:
Yes, but `v1beta1` may not always be available, I believe it can be removed 6 or so versions after deprecation according to: https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-parts-of-the-api . Not an issue right now but just noting there's a chance this stops working on a future Kubernetes version.

Reply:
@Danil-Grigorev If the v1beta1 is only needed for OCP3.7, perhaps we could add a conditional only for that release to use it. This issue I don't think is unique to the nginx role on OCP37 , I assume will also fail in all other tests on the repo. Another idea could be to use a variable for api_version and just change it when we detect an older OCP release (3.7)